src/Popover/Popover.js

```diff
@@ -78,13 +79,22 @@ class Popover extends Component {
             onClickFunctions = chain(this.triggerBody, control.props.onClick);
         }
 
-        const referenceComponent = React.cloneElement(control, {
-            onClick: onClickFunctions,
-            tabIndex: 0,
-            role: 'button',
-            'aria-haspopup': true,
-            onKeyPress: (event) => this.handleKeyPress(event, control, onClickFunctions)
-        });
+        let controlProps = {
+            onClick: onClickFunctions
+        };
+
+        if (!disableKeyPressHandler) {
+            controlProps = {
+                ...controlProps,
+                tabIndex: 0,
+                role: 'button',
+                ariaRole: 'button',
```

Comment:
I'm not sure if we need `aria-role` at all if `role` is present. Also, React converts `ariaRole` to `ariarole`.

Reply:
I've fixed the role handling.